oxidation/libparsec/crates/client_types/src/path.rs

```diff
@@ -0,0 +1,69 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 (eventually AGPL-3.0) 2016-present Scille SAS
+
+use serde::Serialize;
+use std::{
+    fmt::Debug,
+    path::{Path, PathBuf},
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Hash, Serialize)]
+pub struct StrPath(PathBuf);
+
+impl std::str::FromStr for StrPath {
+    // Type Err require display for generator
+    type Err = &'static str;
```

Comment:
use [`Infallible`](https://doc.rust-lang.org/std/convert/enum.Infallible.html)

Reply:
```Shell
error[E0277]: the trait bound `std::convert::Infallible: std::convert::AsRef<str>` is not satisfied
   --> oxidation/bindings/electron/src/meths.rs:21:37
    |
21  |             Err(err) => return (cx).throw_type_error(err),
    |                                     ^^^^^^^^^^^^^^^^ the trait `std::convert::AsRef<str>` is not implemented for `std::convert::Infallible`
    |
note: required by a bound in `neon::context::Context::throw_type_error`
   --> /home/engineer/.cargo/registry/src/github.com-1ecc6299db9ec823/neon-0.10.1/src/context/mod.rs:570:28
    |
570 |     fn throw_type_error<S: AsRef<str>, T>(&mut self, msg: S) -> NeonResult<T> {
    |                            ^^^^^^^^^^ required by this bound in `neon::context::Context::throw_type_error`
    ```

Reply:
🤬 fine

Reply:
Can impl `FromStr` only on `wasm32`

Reply:
```rust
    #[cfg(wasm32)]
    // Type Err require display for generator
    type Err = &'static str;
    #[cfg(not(wasm32))]
    type Err = Infallible;
```
Something like that then ?

Reply:
The error came from electron (native), but it seems overkill no ? Also `StrPath` is used only for the client

Reply:
Ok then